sphinx_needs/filter_common.py

```diff
@@ -300,15 +304,20 @@ def filter_single_need(
     # Get needs external filter data and merge to filter_context
     filter_context.update(app.config.needs_filter_data)
 
-    filter_context["search"] = re.search
+    filter_context["search"] = need_search
     result = False
     try:
         # Set filter_context as globals and not only locals in eval()!
         # Otherwise, the vars not be accessed in list comprehensions.
         if filter_compiled:
-            result = bool(eval(filter_compiled, filter_context))
+            result = eval(filter_compiled, filter_context)
         else:
-            result = bool(eval(filter_string, filter_context))
+            result = eval(filter_string, filter_context)
+        if not isinstance(result, bool):
+            # Raises NeedsInvalidFilter if the result is a string type
+            raise NeedsInvalidFilter(
+                f"Error when evaluating filter: expected output to have True/False but got a string <{result}>"
+            )
```

Comment:
If it is not a bool, this does not logically mean that it will be a string though?